src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    if (this.hidden === true) {
+      this.hidden = false;
+    }
+
+    if (this.hidden === false) {
```

Comment:
Checkout checklist
[CODE KNOWLEDGE] When you need to toggle values, use logical operator !
BAD EXAMPLE:

 if (this.hasPassed === false) {
    this.hasPassed = true;
 } else {
   this.hasPassed = false;
 }
GOOD EXAMPLE:

this.hasPassed = !this.hasPassed